EIPS/eip-5593.md

```diff
@@ -0,0 +1,84 @@
+- If the iframe is first-party to the top-level origin AND the `sandbox attribute is set on the iframe, the provider object MUST be blocked. If the sandbox attribute is set to `sandbox="allow-same-origin"` it MUST be injected for a first party frame.
+    - Note `"allow-same-origin"` does nothing if the iframe is third-party. The case of the third party iframe is dictated by the usage of the `allow` attribute and the Permissions API as defined in the rule above.
+
+## Rationale
+
+By limiting the capabilities of where the Ethereum Provider APIs are being injected we can reduce the surface area of where attacks can be executed.Given the sensitivity of data that's passed to the Ethereum Provider APIs some basic levels of authentication and confidentiality should be met in order to ensure that request data is not being intercepted or tampered with. While there has been attempts to [limit request access via the wallet](./eip-2255.md) interface itself there's not been limitations that have been set to where these Ethereum Provider APIs are expected to be or not be injected. Since the secure contexts web platform API is a well developed boundary that's been recommended by W3C and the fact that the Ethereum Provider APIs are extending the traditional web platform APIs, no other alternative solutions have been considered in order to extend current established prior art.
+
+
+## Backwards Compatibility
+
+Wallet extensions SHOULD consider adding a "developer mode" toggle via a UX so that dApp developers have the capability to disable the insecure context (http) check for the http://localhost:<any-port> origin only in the event that <a href="https://www.w3.org/TR/secure-contexts/#localhost">localhost does not return `true` for secure context</a>. This will allow dApp developers to be able to continue to host dApps on the localhost origin if a browser environment has chosen to not already consider localhost a secure context. Most major browser providers do consider localhost a secure context already. This toggle MUST be set to disabled by default.
+
+## Test Cases
+
+### Required Test Cases
+
+- Top level `http://a.com` -> blocked (insecure/top level)
+- Top level `https://a.com` -> allowed
+- Top level `https://a.com` with `<iframe src="http://a.com/">` -> blocked (insecure first party iframe)
+- Top level `http://a.com` with `<iframe src="https://a.com/">` -> blocked (insecure top level window)
+- Top level `https://a.com` with `<iframe src="https://a.com">` -> allowed
+- Top level `https://a.com` with `<iframe src="https://b.com">` -> blocked (3p iframe without sufficient privileges)
+- Top level `https://b.com` with `<iframe src="http://a.com/">` with `<iframe src="https://b.com">` -> blocked (insecure iframe)
+- Top level `https://b.com` with `<iframe src="https://a.com">` with `<iframe src="https://b.com">` -> blocked (3p iframe without sufficient privileges)
+- Top level `https://a.com` with `<iframe src="https://sub.a.com">` -> allowed
```

Comment:
Why is this allowed? https://a.com and https://sub.a.com are completely different origins. There should be no difference between this and a.com and b.com. We should require the definer to add the `allow="ethereum"` explicitly here.

Reply:
Cory is correct here; I think there was some confusion because Brave considered changing "third party" to "different eTLD+1" at one point because the policy as-written would break ImmutableX (https://github.com/brave/brave-core/pull/13564), but we never actually did that. @kdenhartog do you see any cases where we are considering suborigins as first party for provider injection?

we should probably explicitly state this case in https://github.com/brave/brave-wallet-docs/blob/main/docs/provider-availability.md and check that brave-core behavior matches.

Reply:
actually i see it's already stated here in our docs:
 
> Top level https://a.com with <iframe src="https://b.a.com"> -> blocked (3p)

Reply:
Good catch, I'll double check my notes and tests to see what's going on here. I think what happened was because [example 5 in the spec](https://www.w3.org/TR/secure-contexts/#examples-framed) shows the suborigin as a secure context I was under the impression that this would be allowed.

Reply:
I've confirmed I can't connect in this test case by using this: https://projectopensea.github.io/test-3p-sandbox-iframe/

So this one just needs to be updated to blocked (3p) again

Reply:
This one has been properly updated here now